scarlet/component.py

```diff
@@ -221,31 +242,18 @@ def get_model(self, *parameters):
         if sed is None:
             sed = self.sed
 
-        if shift is None:
-            shift = self.shift
-
         if morph is None:
-            # dont' use self._morph because we could have shift as parameter
-            morph = self._pad_morph(self._shift_morph(shift, self._parameters[1]._data))
-        else:
-            morph = self._pad_morph(self._shift_morph(shift, morph))
+            morph = self._parameters[1]._data
 
-        return sed[:, None, None] * morph[None, :, :]
-
-    def _pad_sed(self, sed):
-        if self.bbox is not None:
-            padded = np.pad(sed, self.pad_width[0], mode="constant", constant_values=0)
-            return padded[self.slices[0]]
-        else:
-            return sed
+        if shift is None:
+            shift = self.shift
 
-    def _pad_morph(self, morph):
-        if self.bbox is not None:
-            padded = np.pad(
-                morph, self.pad_width[1:], mode="constant", constant_values=0
-            )
-            return padded[self.slices[1:]]
-        return morph
+        morph = self._shift_morph(shift, morph)
+        model = sed[:, None, None] * morph[None, :, :]
+        # project the model into frame (if necessary)
+        if frame is not None:
+            model = self.project(model, frame)
```

Comment:
Oh, I see. Have you checked if this is faster than expanding `morph` to full `model_frame` and then doing the outer product with `sed`? Slicing the full model is an operation that isn't aligned well in memory.

Reply:
During optimization `frame` is `None`, so this line doesn't execute anytime we're crunched for speed. This is more for visualizations.